Pipeline/SimplePackageDeploy/SimplePackageDeploy.groovy

```diff
@@ -0,0 +1,287 @@
+ *  5) Reads the BuildReport.json file from the temporary folder.
+ *  6) Identifies all the components to be deployed from the BuildReport.json file.
+ *  7) Identifies deploy types for the components to be deployed from the 
+ *     BuildReport.json file.
+ *  8) Identifies the target libraries based on the deploy type by concatenating the
+ *     --hlq value with the correspoding LLQ value from the JSON string - targetLibLLQMap.
+ *  9) Identifies the copy mode based on the deploy type and corresponding copy mode 
+ *     from the JSON string - copyModeMap.
+ * 10) Copies the components using the copy mode to the respective target libraries.
+ * 11) Deletes the temporary package extract folder after deployment.
+ *
+ * RETURNS:
+ *    Exit Status 0       -  Success
+ *    Exit Status 1       -  Failure
+ *
+ ************************************************************************************/
+
+@Field Properties props = null
+props = parseInput(args)
+
+def startTime = new Date()
+props.startTime = startTime.format("yyyyMMdd.hhmmss.mmm")
+println("** SimplePackageDeploy start at $props.startTime")
+
+// Map of last level dataset qualifier to DBB CopyToPDS CopyMode.
+def copyModeMap = parseJSONStringToMap(props.copyModeMap)
+// Map of last level dataset qualifier to Target PDS last level qualifier.
+def targetLibLLQMap = parseJSONStringToMap(props.targetLibLLQMap)
+
+// Define and create tar file extract directory 
+def tarExtractDirName = "${props.workDir}/DeployFiles_${props.startTime}"
+def tarExtractDir = new File("${tarExtractDirName}")
+
+!tarExtractDir.exists() ?: tarExtractDir.deleteDir()
+tarExtractDir.mkdirs()
+println("** Created tar file extract directory $tarExtractDirName")       
+
+// Check if the tar file provided is existing 
+def tarFile = new File("${props.tarFileName}")
+
+if(!tarFile.exists()){
+    println("** Tar file not found at $tarFile")
+    System.exit(1)
+}
+
+
+// Untar the tar file
+println("** Untar file at $tarFile.")
+
+def processCmd = ["sh", "-c", "tar -C $tarExtractDirName -xvf $tarFile"]
+
+def rc = runProcess(processCmd)
+if (rc == 0) println("Package untar done to $tarExtractDirName")
+else {
+	println("Failed to untar $tarFile")
+	System.exit(1)
+}
+
+
+// Read build report data
+def listDirFiles = new File("$tarExtractDirName")
+def buildReportFlag = 0
+
+// Find the BuildReport.json file for deployment 
+listDirFiles.eachFile {
+   if(it.getName().contains("BuildReport.json")) {
+	   if(!it.exists()){
+		    println("** Build report data at ${it} not found")
+		    System.exit(1)
+	   }
+	   buildReportFlag = 1
+	   deployFromBuildReport(it,tarExtractDirName,targetLibLLQMap,copyModeMap)
+   }
+}
+
+// If BuidlReport.json not found in the build output package
+if(!buildReportFlag){
+    println("** Build report data at $tarExtractDirName/BuildReport.json not found")
+}
+
+if (tarExtractDir.exists()) tarExtractDir.deleteDir()
+println("\nDeleted the temporary folder - ${tarExtractDirName}\n")
```

Comment:
I would rather phrase that as `Cleaning up the temporary folder` . But these are just words

Reply:
Reworded as advised.